torch/csrc/distributed/rpc/python_rpc_handler.cpp

```diff
@@ -26,6 +26,21 @@ PythonRpcHandler::PythonRpcHandler() {
   pySerialize_ = getFunction(module, "serialize");
 }
 
+// In default, PythonRpcHandler will call dec_ref() to clean up python objects.
+// It is found that PythonRpcHandler will dec_ref() null python objects and
+// crashed when program exits in Python 3.5 only (The theory is: the python
+// objects are cleaned up before destructing PythonRpcHandler when program
+// exits in Python 3.5).
+// To avoid PythonRpcHandler destructor to call dec_ref(), explicitly assign
+// the py::handle to be none. This will not have memory leak, as when destruting
+// PythonRpcHandler singleton when program exits, all memories will be cleaned
+// up by OS.
+PythonRpcHandler::~PythonRpcHandler() {
+  pyRunFunction_ = py::none();
+  pyLoadReturnValue_ = py::none();
+  pySerialize_ = py::none();
+}
+
```

Comment:
Won't assigning them to be `py::none()` call `dec_ref` on the existing value?

Reply:
if I understand correctly, assigning py::none() will not trigger dec_ref() call.  in default, when py::object is destroyed, dec_ref() will be called first and then assign it as py::none()?